### PR TITLE
Feature/login

### LINKED ISSUE #10: feat: user should be able to login
After verifying the account, a user should be able to log in.

NB. Seed a verified user for the login feature.

Acceptance criteria
- Proper validation
- Return success or error messages
- Return user details
- Return a JWT token that contains user role (either admin, patient, or consultant), user id, and any other detail deemed necessary. 
- Do not return sensitive information eg password
- Write integration tests
- Include a screenshot of postman request in pull request

### PR DESCRIPTION
**Pull request by lovisgod**
## Description
User should be able to login

The following criteria have been satisfied.
- Proper validation
- Return success or error messages
- Return user details
- Return a JWT token that contains user role (either admin, patient, or consultant), user id, and any other detail deemed necessary. 
- Do not return sensitive information eg password

All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #10  

## How Has This Been Tested?
1.  Clone/Pull/Fetch this branch.
2.  Run `npm install` to install the dependencies.
3.  Run `npm test` to run the tests.
4.  Test the `/api/v1/auth/login` endpoint of the server with postman
5.  In case of a successful login, the user should get a `200` status code and also a `JWT TOKEN`

## Screenshots (if applicable, else remove this line / section)
lgoin_screenshot

## Checklist
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have added necessary inline code documentation
-   [x] I have added tests that prove my fix is effective and that this feature works
-   [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by OvieMudi — CHANGES_REQUESTED** [review]
Great work. However, I think these updates are important:

- EsLint rules: some important rules have been turned off (eg. consistent return, unused-vars, param-reassign). We can turn them back on and restructure our code.

- Input validation: we should have a middleware that validates user input before it gets to the database. Some great validator packages can help with that.

**Review by John-Akins — DISMISSED** [review]
For now I think we're good to go. Any further changes and refactoring can be made later on if necessary. Feature's working perfectly and that's the most important.

File: src/controllers/AuthController.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+      // checking if the user exist
+      const user = email ? await User.findOne({ where: { email } })
+        : await User.findOne({ where: { username } });
+
+      if (!user) return sendErrorResponse(res, 404, 'User Not Found!!');
+
+      // compare password
+      const checkPassword = comparePassword(password, user.dataValues.password);
+      if (!checkPassword) return sendErrorResponse(res, 400, 'Incorrect Password');
```
Comment (adaezeodurukwe) [inline src/controllers/AuthController.js]: tiny detail here, we should not use "incorrect password" as an error message, kindly change this to something like "username or password mismatch"

**Review by adaezeodurukwe — COMMENTED** [review]
please include other details  in the data you're returning like user firstname and lastname. It will be helpful for the frontend

File: src/controllers/AuthController.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+// returns a token for logged in user or during sign up
+const userToken = (user) => {
+  const {
+    email, name, role, uuid
+  } = user;
+  return {
+    token: createToken({
+      uuid,
+      name,
```
Comment (adaezeodurukwe) [inline src/controllers/AuthController.js]: you destructured "name" from the user model but no name is defined there.

File: src/controllers/AuthController.js (hunk 1 above)
Comment (lovisgod) [reply src/controllers/AuthController.js]: Done

File: src/controllers/AuthController.js (hunk 2 above)
Comment (lovisgod) [reply src/controllers/AuthController.js]: Fixed.

**Review by adaezeodurukwe — APPROVED** [review]

**Review by OvieMudi — APPROVED** [review]
LGTM

**Pull request merged**